kubernetes/controller/internal/resolution/workerpool/workerpool.go

```diff
@@ -252,10 +287,47 @@ func (wp *WorkerPool) handleWorkItem(ctx context.Context, logger *logr.Logger, i
 			"duration", duration)
 	}
 
+	// get all requesters AFTER resolution completes but BEFORE cleanup
+	// ensures we capture all requesters that were added during the resolution and the wait for it to be finished
+	requesters := wp.setResult(key, result, err)
+	event := ResolutionEvent{
+		Component:  item.Opts.Component,
+		Version:    item.Opts.Version,
+		Error:      err,
+		Requesters: requesters,
+	}
+
+	// non-blocking send so this worker can be done and pick-up the next thing
+	go func() {
```

Comment:
While these are short-lived, it also means that it could put some pressure on memory on high requester loads.

But if there are high requesters it will put pressure on the even reader. Maaaybe, it's actually better to leave this blocking, because then the workers would eventually fill up and there would be a natural circuit breaker since no other workers can spin up and do work until the even channels are cleaned and new notifications can be sent out. 🤔